src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentActor.scala

```diff
@@ -191,11 +191,10 @@ private class DeploymentActor(
     await(launchQueue.purge(runSpec.id))
 
     val instances = await(instanceTracker.specInstances(runSpec.id))
-    val launchedInstances = instances.filter(_.isActive)
 
-    logger.info(s"Killing all instances of ${runSpec.id}: ${launchedInstances.map(_.instanceId)}")
-    await(Future.sequence(launchedInstances.map(i => instanceTracker.setGoal(i.instanceId, Goal.Decommissioned))))
-    await(killService.killInstances(launchedInstances, KillReason.DeletingApp))
+    logger.info(s"Killing all instances of ${runSpec.id}: ${instances.map(_.instanceId)}")
+    await(Future.sequence(instances.map(i => instanceTracker.setGoal(i.instanceId, Goal.Decommissioned))))
+    await(killService.killInstances(instances, KillReason.DeletingApp))
```

Comment:
I think this is even more in line with how it works in 1.8 so 👍

Reply:
@zen-dog we're sending kill requests for all instances now, including the ones which are reserved or unreachable. However, how does that lead to the instance being expunged? If I'm not mistaken, killing a reserved or unreachable instance that has a reservation will not expunge that instance.

Am I missing something?

Reply:
This fix covers the `Unknown` instance case (as described in the ticket). But, you're correct, that `Unreachable` and `UnreachableUnknonw` are not covered by the fix